app/course/directives/courseTable.js

```diff
@@ -524,6 +524,10 @@ let courseTable = function ($rootScope, $timeout, CourseActionCreators, $compile
               CourseActionCreators.updateSection(section);
             }
           });
+          // Check if any overflowed sections can be updated.  If yes, update them.
+          if (sectionGroup.sections && sectionGroup.sections.length > 0){
+            CourseActionCreators.updateSection();
+          }
```

Comment:
**No changes necessary as getting this to work would entail an irrational amount of work to get the project to recognize modern JS features. This is just a note.**

Optional chaining to the rescue!
```js
if( sectionGroup.sections?.length > 0 ) {
  CourseActionCreators.updateSection();
}
```
No need to short-circuit if sections is null or undefined. The `?.` operator implicitly checks for you and returns `undefined` in this case, which is falsy, so it won't update the section.

Reply:
Dang that's cool.  I did not know that was a thing